test/internal/modules/agreements/lib/reducer.js

```diff
@@ -49,7 +50,7 @@ experiment('internal/modules/agreements/lib/reducer', () => {
         code: 'S127',
         dateSigned: undefined,
         isDateSignedKnown: false,
-        startDate: '2020-04-01'
+        startDate: `${moment().year()}-04-01`
```

Comment:
I'm not certain this will work in some cases - e.g. if today's date is in Jan/Feb/March.

The root of this issue looks like the reducer is not a pure function as it is supposed to be - I think the solution would be to pass the current date as an argument to the action creator.

Reply:
I did something slightly off-script: I have left the dates array the same, but I have made a change such that the financial year is calculated based on today's date, if the date signed is not provided. Curious if you think that's OK.

If not, I can add today's date into the `dates` array - the reason I opted not to do that is because I think it'll break other tests if I do...